nexus/types/src/deployment/execution/utils.rs

```diff
@@ -8,36 +8,57 @@ use nexus_sled_agent_shared::inventory::SledRole;
 use omicron_common::address::{Ipv6Subnet, SLED_PREFIX};
 use omicron_uuid_kinds::SledUuid;
 
-use crate::deployment::{
-    Blueprint, BlueprintZoneDisposition, BlueprintZoneType, blueprint_zone_type,
+use crate::{
+    deployment::{
+        Blueprint, BlueprintZoneDisposition, BlueprintZoneType,
+        blueprint_zone_type,
+    },
+    external_api::views::SledPolicy,
 };
 
 /// The minimal information needed to represent a sled in the context of
 /// blueprint execution.
 #[derive(Debug, Clone)]
 pub struct Sled {
     id: SledUuid,
+    policy: SledPolicy,
     sled_agent_address: SocketAddrV6,
+    repo_depot_port: u16,
     role: SledRole,
 }
 
 impl Sled {
     pub fn new(
         id: SledUuid,
+        policy: SledPolicy,
```

Comment:
I think this is fine, but I'm still a little unsettled that blueprint execution operates on the current-as-read-directly-from-the-database `Sled` properties instead of whatever the properties were when the planner ran. Will this work correctly if a sled's policy has changed (in either direction) w.r.t. whether it matches `SledFilter::TufArtifactReplication` since the blueprint was planned?